File: ruby/lib/exceptions/rcon_no_auth_exception.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2008-2011, Sebastian Staudt

require 'exceptions/steam_condenser_exception'

# This exception class indicates that you have not authenticated yet with the
# game server you're trying to send commands via RCON
#
# @author Sebastian Staudt
# @see GameServer#rcon_auth
# @see GameServer#rcon_exec
class RCONNoAuthException < SteamCondenserException

  # Creates a new `RCONNoAuthException` instance
  def initialize
    super 'Not authenticated yet.'
  end

end

File: php/lib/exceptions/RCONNoAuthException.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2008-2009, Sebastian Staudt
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage Exceptions
 */

require_once STEAM_CONDENSER_PATH . 'exceptions/SteamCondenserException.php';

/**
 * @package Steam Condenser (PHP)
 * @subpackage Exceptions
 */
class RCONNoAuthException extends SteamCondenserException
{
	public function __construct()
	{
		parent::__construct("Not authenticated yet.");
	}
}
?>
